pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,640 @@
+
+    For quantum-circuit specific details, we refer the reader to
+    `Vidal and Theis (2018) <https://arxiv.org/abs/1812.06323>`__ ,
+    `Vidal and Theis (2020) <https://www.frontiersin.org/articles/10.3389/fphy.2020.00297/full>`__ ,
+    `Schuld, Sweke and Meyer (2021) <https://journals.aps.org/pra/abstract/10.1103/PhysRevA.103.032430>`__ ,
+    and
+    `Wierichs, Izaac, Wang and Lin (2021) <https://arxiv.org/abs/2107.12390>`__ .
+    An introduction to the concept of quantum circuits as Fourier series can also be found in
+    the
+    `Quantum models as Fourier series <https://pennylane.ai/qml/demos/tutorial_expressivity_fourier_series.html>`__
+    and
+    `General parameter-shift rules <https://pennylane.ai/qml/demos/tutorial_general_parshift.html>`__
+    demos as well as the
+    `qml.fourier module docstring <https://pennylane.readthedocs.io/en/stable/code/qml_fourier.html>`__ .
+
+    **Example**
+
+    Consider the following QNode:
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit(x, Y):
+            qml.RX(x, wires=0)
+            qml.RY(Y[0], wires=0)
+            qml.RY(Y[1], wires=1)
+            qml.CNOT(wires=[0, 1])
+            qml.RY(5*  Y[1], wires=1)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        x = 0.4
+        Y = np.array([1.9, -0.5])
+        f = 2.3
+
+        circuit_value = circuit(x, Y)
+
+    It has three variational parameters ``x`` (a scalar) and two entries of ``Y``
+    (an array-like).
+    A reconstruction job could then be with respect to the two entries of ``Y``,
+    which enter the circuit with one and six integer frequencies, respectively
+    (see the additional examples below for details on how to obtain the frequency
+    spectrum if it is not known):
+
+    >>> nums_frequency = {"Y": {(0,): 1, (1,): 6}}
+    >>> with qml.Tracker(circuit.device) as tracker:
+    ...     rec = qml.fourier.reconstruct(circuit, {"Y": [(0,), (1,)]}, nums_frequency)(x, Y)
+    >>> rec.keys()
+    dict_keys(['Y'])
+    >>> print(*rec["Y"].items(), sep="\n")
+    ((0,), <function _reconstruct_equ.<locals>._reconstruction at 0x7fbd685aee50>)
+    ((1,), <function _reconstruct_equ.<locals>._reconstruction at 0x7fbd6866eee0>)
+    >>> recon_Y0 = rec["Y"][(0,)]
+    >>> recon_Y1 = rec["Y"][(1,)]
+    >>> np.isclose(recon_Y0(Y[0]), circuit_value)
+    True
+    >>> np.isclose(recon_Y1(Y[1]+1.3), circuit(x, Y+np.eye(2)[1]*1.3)
+    True
+
+    We successfully reconstructed the dependence on the two entries of ``Y`` ,
+    keeping ``x`` and the respective other entry in ``Y`` at their initial values.
+    Let us also see how many executions of the device were used to obtain the
+    reconstructions:
+
+    >>> tracker.totals
+    {'executions': 15}
+
+    For more detailed explanations, usage details and additional examples, see
+    the usage details section below.
+
+    .. UsageDetails::
+
+        *Input formatting*
```

Comment:
Minor, but suggest the subsection headings bold to help them stand out, my eyes tend to glance past them:
```suggestion
        **Input formatting**
```